ntoskrnl/ex/init.c

```diff
@@ -1579,17 +1580,30 @@ Phase1InitializationDiscard(IN PVOID Context)
                                WINDOWS_NT_INFO_STRING_PLURAL :
                                WINDOWS_NT_INFO_STRING,
                                &MsgEntry);
+    if (NT_SUCCESS(MsgStatus))
+    {
+        InfoStringFormat = (PCHAR)MsgEntry->Text;
+    }
+    else
+    {
+        InfoStringFormat = (KeNumberProcessors > 1) ?
+          "%u System Processors [%Iu MB Memory] %Z\r\n" :
+          "%u System Processor [%Iu MB Memory] %Z\r\n";
+    }
 
     /* Get total RAM size */
     Size = MmNumberOfPhysicalPages * PAGE_SIZE / 1024 / 1024;
+    /* Round size up. Assumed to better match actual physical RAM size */
+    if (Size * 1024 * 1024 < MmNumberOfPhysicalPages * PAGE_SIZE)
```

Comment:
Just ALIGN_UP_BY it or something.

Reply:
Well, then do what ALIGN_UP_BY does:
`Size = (MmNumberOfPhysicalPages * PAGE_SIZE + _1M - 1) / _1M;`

Reply:
`ALIGN_UP_BY()` is already available at the 3 places.